.github/workflows/test_coveralls_codestyle.yml

```diff
@@ -1,6 +1,6 @@
 name: CI
 
-on: [push, pull_request]
```

Comment:
What is the reason for this removal?
According to the docs, both of these are needed for coveralls to run on pull requests.  https://github.com/coverallsapp/github-action/
which isn't intuitive, admittedly...

Reply:
As you can see below the build still runs. I removed the `push` to remove duplication of the jobs. In theory, we may have different workflows of push and PR (e.g. only check code style on push) in future. And for now, there's no need to run the workflow twice.